blueocean-jwt/src/main/java/io/jenkins/blueocean/auth/jwt/impl/JwtImpl.java

```diff
@@ -0,0 +1,151 @@
+package io.jenkins.blueocean.auth.jwt.impl;
+
+import com.google.common.collect.ImmutableList;
+import hudson.Extension;
+import hudson.Plugin;
+import hudson.model.User;
+import hudson.remoting.Base64;
+import hudson.tasks.Mailer;
+import io.jenkins.blueocean.auth.jwt.JwkService;
+import io.jenkins.blueocean.auth.jwt.JwtAuthenticationService;
+import io.jenkins.blueocean.auth.jwt.JwtToken;
+import io.jenkins.blueocean.commons.ServiceException;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.acegisecurity.Authentication;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.security.interfaces.RSAPublicKey;
+import java.util.Collections;
+import java.util.UUID;
+
+/**
+ * @author Vivek Pandey
+ */
+@Extension
+public class JwtImpl extends JwtAuthenticationService {
+
+    private static int DEFAULT_EXPIRY_IN_SEC = 1800;
+    private static int DEFAULT_MAX_EXPIRY_TIME_IN_MIN = 480;
+    private static int DEFAULT_NOT_BEFORE_IN_SEC = 30;
+
+    @Override
+    public JwtToken getToken(@Nullable @QueryParameter("expiryTimeInMins") Integer expiryTimeInMins, @Nullable @QueryParameter("maxExpiryTimeInMins") Integer maxExpiryTimeInMins) {
+        String t = System.getProperty("EXPIRY_TIME_IN_MINS");
+        long expiryTime=DEFAULT_EXPIRY_IN_SEC;
+        if(t!= null){
+            expiryTime = Integer.parseInt(t);
+        }
+
+        int maxExpiryTime = DEFAULT_MAX_EXPIRY_TIME_IN_MIN;
+
+        t = System.getProperty("MAX_EXPIRY_TIME_IN_MINS");
+        if(t!= null){
+            maxExpiryTime = Integer.parseInt(t);
+        }
+
+        if(maxExpiryTimeInMins != null){
+            maxExpiryTime = maxExpiryTimeInMins;
+        }
+        if(expiryTimeInMins != null){
+            if(expiryTimeInMins > maxExpiryTime) {
+                throw new ServiceException.BadRequestExpception(
+                    String.format("expiryTimeInMins %s can't be greated than %s", expiryTimeInMins, maxExpiryTime));
+            }
+            expiryTime = expiryTimeInMins * 60;
+        }
+
+        Authentication authentication = Jenkins.getInstance().getAuthentication();
+
+        if(authentication == null){
+            throw new ServiceException.UnauthorizedException("Unauthorized: No login session found");
```

Comment:
@ivan there will be no login in the ATH ... will this cause things to blow up there then?